### PR TITLE
migrate to use mapbox-java3.0

### PR DESCRIPTION
**Pull request by osana**
migrated to use mapbox-java3.0
old 2.2.9 telemetry is still used though

closes #10920

### CODE REVIEW

**Comment by osana** [comment]
This should be ready for review. In the test app. **Query Source Features** and **Query feature properties** seem to work. A couple of points to consider:

**Comment by osana** [comment]
Still have one remaining crash that I cannot figure out.
@tobrun could you have  a look. 
It crashes in test app . styles: FillExtrusions 

It crashes while executing in 
FillExtrusionActivity.onCreate()

whiel doing :  GeoJsonSource source = new GeoJsonSource("extrusion-source", domTower);
It is probably something very stupid . The problem seems to be while calling 
  nativeSetGeometry(geometry);  

It crashes pretty reliably.

I see that Navigation repo has mapboxSdk 5.3 and mapbox-java3.0 which is bad match to have at the moment so it is important to resolve this.  I think Navigation people  just got luck that it works.

cc @tobrun

File: platform/android/MapboxGLAndroidSDK/build.gradle [hunk 1]
```diff
@@ -3,6 +3,7 @@ apply plugin: 'com.android.library'
 dependencies {
     api dependenciesList.mapboxAndroidTelemetry
     api dependenciesList.mapboxJavaGeoJSON
+    api dependenciesList.mapboxJavaTurf
```
Comment (tobrun) [inline platform/android/MapboxGLAndroidSDK/build.gradle]: what is the reason why we need this in the SDK?

File: platform/android/src/geojson/conversion/geometry.hpp [hunk 2]
```diff
@@ -112,57 +115,109 @@ class GeometryEvaluator {
...
+//        jni::jarray<jni::jobject>& jarray = jni::NewObjectArray(env, value.size(), *javaClass);
+//
+//        for(size_t i = 0; i < value.size(); i = i + 1) {
+//            jni::LocalObject<jni::jarray<jni::jobject>> coordinates = jni::NewLocalObject(env, toGeoJsonCoordinates(env, value.at(i)));
+//            jni::SetObjectArrayElement(env, jarray, i, coordinates.get());
+//        }
+//
+//        return &jarray;
+//    }
+
```
Comment (tobrun) [inline platform/android/src/geojson/conversion/geometry.hpp]: commented code above

**Comment by osana** [comment]
Updated PR with requested changes
@tobrun

File: platform/android/src/geojson/polygon.cpp [hunk 3]
```diff
@@ -12,19 +12,19 @@ mapbox::geojson::polygon Polygon::convert(jni::JNIEnv &env, jni::Object<Polygon>
...
 }
 
-mapbox::geojson::polygon Polygon::convert(jni::JNIEnv &env, jni::Object<java::util::List/*<java::util::List<Position>>*/> jPositionListsList) {
+mapbox::geojson::polygon Polygon::convert(jni::JNIEnv &env, jni::Object<java::util::List/*<java::util::List<Point>>*/> jPointListsList) {
     mapbox::geojson::polygon polygon;
 
-    if (jPositionListsList) {
-        auto multiLine = MultiLineString::convert(env, jPositionListsList);
+    if (jPointListsList) {
+        auto multiLine = MultiLineString::convert(env, jPointListsList);
```
Comment (osana) [inline platform/android/src/geojson/polygon.cpp]: @tobrun  could it be that DeleteLocalRef() for multiline was and is missing here between lines 31 & 32

Comment (tobrun) [reply platform/android/src/geojson/polygon.cpp]: yes it seems that we aren't DeleteLocalRef on `jPointListsList ` as we are doing in the other classes eg. https://github.com/mapbox/mapbox-gl-native/blob/0ee6c2c5c0fc4303ff3128890eef1ad3db471deb/platform/android/src/geojson/multi_point.cpp#L17. Could you add that?

Comment (osana) [reply platform/android/src/geojson/polygon.cpp]: @tobrun Looks like a false alarm on my side:
``` 
/src/platform/android/src/geojson/polygon.cpp:32:34: error: no viable conversion from 'mapbox::geometry::multi_line_string<double, std::vector>' to 'jni::jobject *'
          jni::DeleteLocalRef(env, multiLine);
                                   ^~~~~~~~~```

**Pull request merged**